packages/jsii-config/lib/schema.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Structure of jsii configuration in package.json
+ */
+export interface JsiiConfig {
+  outdir: string;
+  versionFormat: 'full' | 'short';
+  targets: {
+    java?: {
+      package: string;
+      maven: {
+        groupId: string;
+        artifactId: string;
+        versionSuffix?: string;
+      };
+    };
+    python?: {
+      module: string;
+      distName: string;
+    };
+    dotnet?: {
+      namespace: string;
+      packageId: string;
+      iconUrl?: string;
+      versionSuffix?: string;
+      signAssembly?: boolean;
+      assemblyOriginatorFile?: string;
+    };
+  };
+  metadata?: {
+    [key: string]: any;
+  };
+}
+
+/*
+ * Type of fields used in schema to prompt users.
+ */
+interface Field {
+  type: string;
+  message: string;
+  default?: any;
+  choices?: string[];
+  when?: (answers: any) => boolean;
+  validate?: (val: any, answers: any) => boolean | string;
+  filter?: (val: any) => any | void;
+}
+
+interface NestedField {
+  [key: string]: SchemaField;
+}
+
+type SchemaField = Field | NestedField;
+
+/*
+ * ConfigSchema
+ * 
+ * Requires 'targets' key with available Jsii language targets.
+ */
+export interface ConfigSchema {
+  [key: string]: SchemaField;
+  targets: SchemaField;
+}
+
+/*
+ * Builds a function to ask for target specific values when enabled
+ */
+const targetEnabled = (target: string) => (answers: any) => {
+  return Boolean(answers.jsiiTargets.includes(target));
+};
+
+/*
+ * Validates that user input has length
+ */
+function hasLength(val: string): true | string {
+  const isValid = Boolean(val) && val.trim().length > 0;
+  if (isValid) {
+    return true;
+  }
+  return 'Please enter a value';
+}
+
+/*
+ * filter out values from answers that are allowed to be undefined
+ */
+function filterEmpty(val: string): string | void {
+  if (typeof val === 'string' && val.length === 0) {
+    return undefined;
+  }
+  return val;
+}
+
+/*
+ * Schema that questions are built from.
+ *
+ * Name values for questions are built from parent keys.
+ */
+const schema: ConfigSchema = {
+  outdir: {
+    type: 'input',
+    message: 'output directory for typescript compiler',
+    default: 'dist',
+    validate: hasLength
+  },
+  versionFormat: {
+    type: 'list',
+    message: 'Version format, default is full',
```

Comment:
Can we add a description to the user explaining what this is? I don't even know...